Článek o organizaci sprintu na východě (#88)
pyvec/blog.python.cz: content/2024-04-29_organizujeme-sprinty-a-ty-muzes-taky.md

Title: Organizujeme sprinty. A ty můžeš taky!
Date: 2024-04-29 11:33:00
Modified: 2024-04-29 11:33:00
Author: Lumír Balhar
Twitter: lumirbalhar
Gravatar: frenzy.madness@gmail.com
About: Lumír pracuje jako senior software inženýr a ve volném čase organizuje komunitní srazy, kurzy a workshopy, vyučuje první pomoc a dělá dobrovolného hasiče.

Lidé spojení určitým společným zájmem se rádi potkávají. V Python komunitě je přiležitostí mnoho – konference ([PyConCZ](https://cz.pycon.org/), [EuroPython](https://europython.eu)), [srazy](https://pyvo.cz/), [kurzy](https://pyladies.cz/), [workshopy](http://pyworking.cz/), … a také sprinty – a o jednom takovém bude tento článek.

> Ve více lidech se to lépe táhne a tak, pokud chcete komunitu posouvat kupředu, je nejlepěí si na to dedikovat kousek volného času, někde se sejít a společně udělat kousek užitečné práce. To je nejčastější motivace k uspořádání sprintu. Sprint je z akcí výše jedna z mála, kterou dokážeš připravit i v jednom člověkovi.

Reviewer: ```suggestion
Ve více lidech se to lépe táhne a tak, pokud chcete komunitu posouvat kupředu, je nejlepší si na to dedikovat kousek volného času, někde se sejít a společně udělat kousek užitečné práce. To je nejčastější motivace k uspořádání sprintu. Sprint je z akcí výše jedna z mála, kterou dokážeš připravit i v jednom člověku.
```
[edit: nejlepěí→nejlepší - @encukou]

> Myšlenka vrátit se k tradici sprintů na východě už tu nějakou dobu byla. Poslední sprint se [konal v roce 2019](https://blog.python.cz/Sprint-na-vychode) před [PyConem v Ostravě](https://blog.python.cz/PyConCZ_wrap_up_summary). No a tak stačilo, aby se pár zájemců ozvalo a z myšlenky se začal tvořit plán.

Reviewer: ```suggestion
Myšlenka vrátit se k tradici sprintů na východě už tu nějakou dobu byla. Poslední sprint se [konal v roce 2019](https://blog.python.cz/Sprint-na-vychode) před [PyConem v Ostravě](https://blog.python.cz/PyConCZ_wrap_up_summary). No a tak stačilo, aby se pár zájemců ozvalo, a z myšlenky se začal tvořit plán.
```

V kalendáři jsme našli věc vzácnou jako kvalitní dokumentace – volný víkend – a plánování mohlo začít. Nejdříve je třeba zjistit zájem. Dům je sice relativně velký, ale místa a matrací není neomezeně. Neznamená to, že bychom omezovali počet účastníků, ale kdo se registruje později bude muset počítat s menším komfortem nebo třeba se stanem na zahradě. Proto je lepší dělat sprint za hezkého počasí nebo si sehnat dostatečně velké prostory. Naštěstí si někteří myslí, že Frýdek-Místek už je východní Ukrajina a tak se nám ještě nestalo, že bychom se nevešli.

> Mezi zájemci o sprint mohou být i cizí lidé a proto je třeba detaily dobře komunikovat. Máme doma taková a maková zvířata, přespat se u nás dát tak či onak a doprava je možná těmito X způsoby. To jsou všechno vlastnosti, které rozhodují o tom, kdo bude ochoten se sprintu zúčastnit. U nás je např. třeba počítat se psem vycvičeným k tulení. Obráceně je také nezbytné zjistit něco málo o účastnících. Kdo má jaké možnosti dopravy a plány se na místo dostat. Dietní omezení se pak snadno řeší společným nákupem. Vše popsané se dá jednoduše oboustranně komunikovat jedním google dokumentem či tabulkou.

Reviewer: ```suggestion
Mezi zájemci o sprint mohou být i cizí lidé, a proto je třeba detaily dobře komunikovat. Máme doma taková a maková zvířata, přespat se u nás dát tak či onak, a doprava je možná těmito X způsoby. To jsou všechno vlastnosti, které rozhodují o tom, kdo bude ochoten se sprintu zúčastnit. U nás je např. třeba počítat se psem vycvičeným k tulení. Obráceně je také nezbytné zjistit něco málo o účastnících. Kdo má jaké možnosti dopravy a plány se na místo dostat. Dietní omezení se pak snadno řeší společným nákupem. Vše popsané se dá jednoduše oboustranně komunikovat jedním Google dokumentem, či tabulkou.
```

Jak se přiblížil den D, bylo třeba skočit na nákup a sehnat vše, co by mohli účastníci potřebovat k přežití. Protože některé věci jsem v životě nekupoval, jezdil jsem po obchodním domě chvílemi dokola jako pilot NASCAR, ale nakonec se povedlo vše sehnat. Náklady na sprint u nás nepadají na bedra hosta, ale buď se sežene sponzor, nebo máme možnost [požádat o grant](https://docs.pyvec.org/operations/support-money.html#) naši neziskovku [Pyvec](https://pyvec.org/).

> S menšími či většími potížemi se všichni zvládli dostat na místo. Někomu stačila rada z oblasti místopisu, jiný potřeboval vyzvednout na nádraží. Ať tak nebo tak, v pátek odpoledne už jsme všichni byli bezpečně na místě. Proběhla nezbytná prohlídka domu, rychlá instruktáž (tady spíte, tady je toaleta, tady je jídlo, heslo na wifi, …) a mohli jsme začít debatovat. Pátek byl pro většinu pracovní den a tak se k počítačům nikomu nechtělo. Navíc v konkurenci venkovního táboráku.

Reviewer: ```suggestion
S menšími či většími potížemi se všichni zvládli dostat na místo. Někomu stačila rada z oblasti místopisu, jiný potřeboval vyzvednout na nádraží. Ať tak nebo tak, v pátek odpoledne už jsme všichni byli bezpečně na místě. Proběhla nezbytná prohlídka domu, rychlá instruktáž (tady spíte, tady je toaleta, tady je jídlo, heslo na wifi, …) a mohli jsme začít debatovat. Pátek byl pro většinu pracovní den, a tak se k počítačům nikomu nechtělo. Navíc v konkurenci venkovního táboráku.
```

> To hlavní se dělo v sobotu. Ráno snídaně v loveckém stylu – ulov si co chceš – a pak hurá do práce. Kdo neměl úkoly vyyšlené, tomu byly přiděleny. Můj hlavní úkol byl udržet pracující v maximální pohodě. Jídlo, pití, elektřina, připojení k internetu, zobání, canisterapie, objednávka oběda, hudba, plánování večera, nákup potravin a tak dále. Moc jsem toho na počítači neudělal, ale staral jsem se o to, aby toho hodně udělali jiní. Chvílemi se intenzivně programovalo, pak diskutovalo a plánovalo a i v relativně malém počtu lidí se sloupec Done na našem kanban boardu plnil velmi rychle.

Reviewer: ```suggestion
To hlavní se dělo v sobotu. Ráno snídaně v loveckém stylu – ulov si co chceš – a pak hurá do práce. Kdo neměl úkoly vymyšlené, tomu byly přiděleny. Můj hlavní úkol byl udržet pracující v maximální pohodě. Jídlo, pití, elektřina, připojení k internetu, zobání, canisterapie, objednávka oběda, hudba, plánování večera, nákup potravin a tak dále. Moc jsem toho na počítači neudělal, ale staral jsem se o to, aby toho hodně udělali jiní. Chvílemi se intenzivně programovalo, pak diskutovalo a plánovalo, a i v relativně malém počtu lidí se sloupec Done na našem kanban boardu plnil velmi rychle.
```

Odpolední pracovní nasazení přešlo volně k večernímu posezení u táboráku, opékání buřtů, grilování sýrů a diskusím na různá témata.

> V neděli se někteří museli vydat na cestu k domovu časně ráno, takže jsem jim pomohl se dopravit na vlak. Něco málo se ještě stihlo udělat či alespoň připravit a mým hlavním úkolem bylo zajistit snídani, úklid a následně všechny bezpečně dopravit na vlak.

Reviewer: ```suggestion
V neděli se někteří museli vydat na cestu k domovu časně ráno, takže jsem jim pomohl se dopravit na vlak. Něco málo se ještě stihlo udělat, či alespoň připravit, a mým hlavním úkolem bylo zajistit snídani, úklid a následně všechny bezpečně dopravit na vlak.
```

> Povedlo se! Co se na začátku zdálo jako stres a spousta práce nakonec uteklo jako voda a na konci sice bylo unavené tělo, ale velmi spokojená duše.

Reviewer: ```suggestion
Povedlo se! Co se na začátku zdálo jako stres a spousta práce, nakonec uteklo jako voda, a na konci sice bylo unavené tělo, ale velmi spokojená duše.
```
